diff -Nur a/xbmc/cores/DllLoader/DllLoader.h b/xbmc/cores/DllLoader/DllLoader.h
--- a/xbmc/cores/DllLoader/DllLoader.h	2015-12-14 23:07:12.000000000 +0000
+++ b/xbmc/cores/DllLoader/DllLoader.h	2016-02-09 22:41:51.217347550 +0000
@@ -23,7 +23,7 @@
 #include "coffldr.h"
 #include "LibraryLoader.h"
 
-#if defined(__linux__) && !defined(__powerpc__) && !defined(__arm__) && !defined(__mips__)
+#if defined(__linux__) && !defined(__powerpc__) && !defined(__arm__) && !defined(__mips__) && !defined(__aarch64__)
 #define USE_LDT_KEEPER
 #include "ldt_keeper.h"
 #endif
diff -Nur a/xbmc/cores/DllLoader/ldt_keeper.c b/xbmc/cores/DllLoader/ldt_keeper.c
--- a/xbmc/cores/DllLoader/ldt_keeper.c	2015-12-14 23:07:12.000000000 +0000
+++ b/xbmc/cores/DllLoader/ldt_keeper.c	2016-02-09 22:41:51.221347549 +0000
@@ -19,7 +19,7 @@
  */
 
 //#ifndef __powerpc__
-#if !defined(__powerpc__) && !defined(__ppc__) && !defined(__arm__) && !defined(__mips__)
+#if !defined(__powerpc__) && !defined(__ppc__) && !defined(__arm__) && !defined(__mips__) && !defined(__aarch64__)
 
 #include "ldt_keeper.h"
 
diff -Nur a/xbmc/cores/VideoRenderers/LinuxRendererGL.h b/xbmc/cores/VideoRenderers/LinuxRendererGL.h
--- a/xbmc/cores/VideoRenderers/LinuxRendererGL.h	2015-12-14 23:07:12.000000000 +0000
+++ b/xbmc/cores/VideoRenderers/LinuxRendererGL.h	2016-02-09 22:41:51.221347549 +0000
@@ -323,7 +323,7 @@
 
 
 inline int NP2( unsigned x ) {
-#if defined(TARGET_POSIX) && !defined(__POWERPC__) && !defined(__PPC__) && !defined(__arm__) && !defined(__mips__)
+#if defined(TARGET_POSIX) && !defined(__POWERPC__) && !defined(__PPC__) && !defined(__arm__) && !defined(__mips__) && !defined(__aarch64__)
   // If there are any issues compiling this, just append a ' && 0'
   // to the above to make it '#if defined(TARGET_POSIX) && 0'
 
diff -Nur a/xbmc/cores/dvdplayer/DVDCodecs/Video/AMLCodec.cpp b/xbmc/cores/dvdplayer/DVDCodecs/Video/AMLCodec.cpp
--- a/xbmc/cores/dvdplayer/DVDCodecs/Video/AMLCodec.cpp	2015-12-14 23:07:12.000000000 +0000
+++ b/xbmc/cores/dvdplayer/DVDCodecs/Video/AMLCodec.cpp	2016-02-09 22:42:48.389333636 +0000
@@ -1590,7 +1590,7 @@
         am_private->gcodec.param = (void*)(EXTERNAL_PTS | SYNC_OUTSIDE);
       break;
   }
-  am_private->gcodec.param = (void *)((unsigned int)am_private->gcodec.param | (am_private->video_rotation_degree << 16));
+  am_private->gcodec.param = (void *)((unsigned long)am_private->gcodec.param | (am_private->video_rotation_degree << 16));
 
   // translate from generic to firemware version dependent
   m_dll->codec_init_para(&am_private->gcodec, &am_private->vcodec);
diff -Nur a/xbmc/guilib/GUIFontTTFGL.cpp b/xbmc/guilib/GUIFontTTFGL.cpp
--- a/xbmc/guilib/GUIFontTTFGL.cpp	2015-12-14 23:07:12.000000000 +0000
+++ b/xbmc/guilib/GUIFontTTFGL.cpp	2016-02-09 22:41:51.221347549 +0000
@@ -232,7 +232,7 @@
       glUniformMatrix4fv(modelLoc, 1, GL_FALSE, glMatrixModview.Get());
 
       // Bind the buffer to the OpenGL context's GL_ARRAY_BUFFER binding point
-      glBindBuffer(GL_ARRAY_BUFFER, (GLuint) m_vertexTrans[i].vertexBuffer->bufferHandle);
+      glBindBuffer(GL_ARRAY_BUFFER, (unsigned long) m_vertexTrans[i].vertexBuffer->bufferHandle);
 
       // Do the actual drawing operation, split into groups of characters no
       // larger than the pre-determined size of the element array
diff -Nur a/xbmc/threads/Atomics.cpp b/xbmc/threads/Atomics.cpp
--- a/xbmc/threads/Atomics.cpp	2015-12-14 23:07:12.000000000 +0000
+++ b/xbmc/threads/Atomics.cpp	2016-02-09 22:41:51.221347549 +0000
@@ -106,7 +106,7 @@
 ///////////////////////////////////////////////////////////////////////////
 long long cas2(volatile long long* pAddr, long long expectedVal, long long swapVal)
 {
-#if defined(__ppc__) || defined(__powerpc__) || defined(__arm__)// PowerPC and ARM
+#if defined(__ppc__) || defined(__powerpc__) || defined(__arm__) || defined(__aarch64__) // PowerPC and ARM
 // Not available/required
 // Hack to allow compilation
   throw "cas2 is not implemented";
diff -Nur a/xbmc/utils/MathUtils.h b/xbmc/utils/MathUtils.h
--- a/xbmc/utils/MathUtils.h	2015-12-14 23:07:12.000000000 +0000
+++ b/xbmc/utils/MathUtils.h	2016-02-09 22:41:51.221347549 +0000
@@ -35,7 +35,7 @@
 #if defined(__ppc__) || \
     defined(__powerpc__) || \
     defined(__mips__) || \
-    defined(__arm__)
+    defined(__arm__) || defined(__aarch64__)
   #define DISABLE_MATHUTILS_ASM_ROUND_INT
 #endif
 
